File: B+Track/MXLWeek.m
//
//  MXLWeek.m
//  B+Track
//
//  Created by Fitti Weissglas on 12/06/2014.
//  Copyright (c) 2014 MXL. All rights reserved.
//

#import "MXLWeek.h"

@implementation MXLWeek

- (id) initWithWeekNo:(NSString *)weekno displayName:(NSString*)displayName {
    self = [super init];
    if (self) {
        self.weekno = weekno;
        self.displayName = displayName;
    }
    
    return self;
}

- (id) initWithWeekNo:(NSString *)weekno startDate:(NSDate*)startDate endDate:(NSDate*)endDate {
    self = [super init];
    if (self) {
        self.weekno = weekno;
        self.startDate = startDate;
        self.endDate = endDate;
        self.displayName = [NSString stringWithFormat:@"Week ending %@", endDate];
    }
    
    return self;
}

@end

File: B+Track/MXLWeek.swift
//
//  MXLWeek.swift
//  BPlus_Track
//
//  Created by Fitti Weissglas on 04/10/2014.
//  Copyright (c) 2014 MXL. All rights reserved.
//

import Foundation

@objc
public class MXLWeek : NSObject {
    
    public var weekno: String;
    public var displayName: String;
    public var startDate: NSDate?;
    public var endDate: NSDate?;
    
    public init(weekNo: String, displayName: String) {
        self.weekno = weekNo;
        self.displayName = displayName;
    }
    
    
    public init(weekNo: String, startDate: NSDate, endDate: NSDate) {
        self.weekno = weekNo;
        self.startDate = startDate;
        self.endDate = endDate;
        self.displayName = NSString(format:"SWIFT! Week ending %@", endDate);
    }
}

